### PR TITLE
duck 4.7.7

### PR DESCRIPTION
**Pull request by dkocher**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

**Comment by nijikon** [comment]
Please squash to one commit.

File: Library/Formula/duck.rb [hunk 1]
```diff
@@ -16,11 +16,13 @@ class Duck < Formula
   depends_on :java => ["1.8+", :build]
   depends_on :xcode => :build
   depends_on "ant" => :build
```
Comment (dunn) [inline Library/Formula/duck.rb]: Ant is still required, alongside Maven?

Comment (dkocher) [inline Library/Formula/duck.rb]: Yes.

**Comment by dunn** [comment]
```

**Comment by mistydemeo** [comment]
We normally pass `-headerpad_max_install_names` to the linker in order to ensure there's enough room to rewrite install_names; superenv is meant to ensure the flag gets passed.

**Comment by dkocher** [comment]
The issue is that `dylib` libraries in `duck/4.7.7.18995/libexec/Contents/MacOS/Runtime.jre/Contents/Home/lib/*` are binary dependencies we do not build ourselves.

I think we had these warnings in previous releases as well. Is this a blocker?

**Comment by mikemcquaid** [comment]
> Is this a blocker?

Yes.

Are you shipping a full JRE with Duck now?

**Comment by dkocher** [comment]
We always shipped with a JRE.

**Comment by dkocher** [comment]
What is the intent to run `install_name_tool` on the libraries? We already do this in our build and the paths should be right.

**Comment by mikemcquaid** [comment]
We need to run it for our binary packaging. Did you definitely `brew install` this locally before submitting this PR?

**Comment by dkocher** [comment]
@mikemcquaid I installed it locally. The `Failed to fix install names` error does not prevent it from installing.

**Comment by dkocher** [comment]
@mikemcquaid How should we proceed? From my viewpoint there is no need to fiddle with {{install_name_tool}} as all references in libraries are relative to the bundle.

**Comment by mikemcquaid** [comment]
@dkocher This needs to be built such that our `install_name_tool` invocations do not fail.

**Comment by dkocher** [comment]
@mikemcquaid I still fail to see why you want to change the install name. `otool -L` says the library that gives the error is linked with
```
@rpath/libbci.dylib (compatibility version 1.0.0, current version 1.0.0)
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 169.3.0)
```
This is perfectly fine and must not be altered.

**Comment by mikemcquaid** [comment]
@dkocher We do it for every piece of software in Homebrew to manage our relocatable binary packages. Whether it's needed or not: this PR cannot be merged as-is. The simplest solution is to fix your code so our `install_name_tool` can be run. The alternative is to close this pull request.

**Comment by mikemcquaid** [comment]
@dkocher @G-Square We need to figure something out here or remove this software from Homebrew.

**Comment by mikemcquaid** [comment]
And to be more explicit: this needs to be fixed to not break on Homebrew's `install_name_tool` invocations.

**Comment by apjanke** [comment]
Closing this due to inactivity/lack of response. If you'd still like to pursue this, feel free to respond to the previous comments/questions and re-open on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core).

**Pull request closed without merging**